clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitResponse.java

```diff
@@ -95,13 +94,8 @@ public OffsetCommitResponseData data() {
 
     @Override
     public Map<Errors, Integer> errorCounts() {
-        List<Errors> errors = new ArrayList<>();
-        for (OffsetCommitResponseTopic topic : data.topics()) {
-            for (OffsetCommitResponsePartition partition : topic.partitions()) {
-                errors.add(Errors.forCode(partition.errorCode()));
-            }
-        }
-        return errorCounts(errors);
+        return errorCounts(data.topics().stream().flatMap(topicResult -> topicResult.partitions().stream())
```

Comment:
Is there a reason why we are not doing the map within the flatMap?

Reply:
No _good_ reason. Good spot.